includes/Core/Authentication/Google_Proxy.php

```diff
@@ -282,23 +298,36 @@ public function exchange_site_code( $site_code, $undelegated_code ) {
 			)
 		);
 
-		if ( is_wp_error( $response ) ) {
-			throw new Exception( $response->get_error_code() );
-		}
-
-		$raw_body      = wp_remote_retrieve_body( $response );
-		$response_data = json_decode( $raw_body, true );
-
-		if ( ! $response_data || isset( $response_data['error'] ) ) {
-			throw new Exception(
-				isset( $response_data['error'] ) ? $response_data['error'] : 'failed_to_parse_response'
-			);
+		if ( is_wp_error( $response_data ) ) {
+			return $response_data;
 		}
 
 		if ( ! isset( $response_data['site_id'], $response_data['site_secret'] ) ) {
-			throw new Exception( 'oauth_credentials_not_exist' );
+			return new WP_Error( 'oauth_credentials_not_exist' );
```

Comment:
I know this was missing before too, but let's add a corresponding message here now.

Reply:
Updated.